src/vs/platform/extensions/electron-main/directMainProcessExtensionHostStarter.ts

```diff
@@ -0,0 +1,29 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ExtensionHostStarter, IPartialLogService } from 'vs/platform/extensions/node/extensionHostStarter';
+import { ILifecycleMainService } from 'vs/platform/lifecycle/electron-main/lifecycleMainService';
+import { ILogService } from 'vs/platform/log/common/log';
+
+export class DirectMainProcessExtensionHostStarter extends ExtensionHostStarter {
```

Comment:
This file does not seem to be used anymore but I understand that you probably want to keep it for testing the starter without worker.

Reply:
:+1: Yes, there is also some code in the shared process that sets up an `IExtensionHostStarter` service that can be removed once we get confidence that spawning from main is the best way forward.

Reply:
I've created https://github.com/microsoft/vscode/issues/137025 to remember to do it.